src/client/datascience/jupyter/debuggerVariables.ts

```diff
@@ -243,25 +248,45 @@ export class DebuggerVariables extends DebugLocationTracker
         try {
             // Run our dataframe scripts only once per session because they're slow
             const key = this.debugService.activeDebugSession?.id;
-            if (key && !this.importedIntoKernel.has(key)) {
-                await this.evaluate(DataFrameLoading.DataFrameSysImport);
-                await this.evaluate(DataFrameLoading.DataFrameInfoImport);
-                await this.evaluate(DataFrameLoading.DataFrameRowImport);
-                await this.evaluate(DataFrameLoading.VariableInfoImport);
-                this.importedIntoKernel.add(key);
+            if (key && !this.importedDataFrameScriptsIntoKernel.has(key)) {
```

Comment:
Just wanted to make sure you tested this with interactive debugging and run by line?

Reply:
@rchiodo I tested it with interactive cell debugging, but not RBL yet. I'll check that out quick.

Reply:
Yeah, looks good here (I'd already run the cell once, so the vars were already there):
![image](https://user-images.githubusercontent.com/812783/100785731-ecd6b500-33c5-11eb-9da0-9b2b66a6708e.png)

Reply:
And dataframes open while debugging too?

Reply:
@rchiodo Bleh, I'm glad you asked this. I think it was working in the above example as the variables had already been seen before in a non-debug context.

I don't think this is all working right for the debugger right now. I think that evaluate just takes an expression, so it's only getting one function if the script file contents are passed in.

I think that I can still use this approach for kernelVariables (as those work remote) then convert debugger back to using an import statement. Debugger is only local, so that should be acceptable. The import is a single expression which is probably why it works with evaluate.

Reply:
@rchiodo Not sure if you want to take another look, but I reverted debugger back to using imports. Pretty close to the old code only we just have one file that we import now instead of two for the data frame functions. Checked the following scenarios:
Local IW Vars + DataFrame opening
Local Notebook Vars + DataFrame opening
Local IW Cell Debugging Vars + DataFrame opening
Local Notebook Run By Line Vars + DataFrame opening
Remote IW and Notebook Vars + DataFrame opening

Reply:
Did you see Joyce's Team's comment about the view in variable menu? This might not work in remote situations for that too.

Reply:
@rchiodo Did you mean on GitHub, not Teams? I commented on a thread on GitHub.

If I was reading that right it's more an issue that the sys path might not work for the situation of opening a data frame viewer from a remote debugging session. Since the debuggerVariables path hasn't really changed in this PR I was figuring that I'd still push this PR to fix the non-debugging cases, then pick that up next if it's considered a must-fix.

Was that right? Or were you concerned about this PR regressing something there?

Might have to do something more interesting if that's the case. I think that I could push each individual function over in an evaluate, though that sounds kinda ugly.

Reply:
Yeah sorry it was this issue here:
https://github.com/microsoft/vscode-jupyter/issues/4065

I agree this isn't necessary now. (especially if this bug is must fix)